### PR TITLE
scheduled removal of auto_move_data decorator

### PR DESCRIPTION
**Pull request by Tshimanga**
## What does this PR do?

Removes `auto_move_data` decorator schedule for removal in 1.5

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/core/test_decorators.py [hunk 1]
```diff
@@ -11,38 +11,8 @@
...
-    model.prepare_data()
-    loader = model.train_dataloader()
-    x = next(iter(loader))
-
-    # test that data on source device gets moved to destination device
-    x = x.to(src_device)
-    assert model(x).device == dest_device, "Automoving data to same device as model failed"
+def test_parameter_validation():
+    # TODO: Fill out this test
+    assert True
```
Comment (Tshimanga) [inline tests/core/test_decorators.py:18]: This file is empty otherwise so I added a stub for the other untested decorator rather than deleting the file.

File: tests/overrides/test_data_parallel.py [hunk 2]
```diff
@@ -151,7 +151,6 @@ def __init__(self):
             super().__init__()
             self.layer = nn.Linear(2, 3)
 
-        @auto_move_data
```
Comment (Tshimanga) [inline tests/overrides/test_data_parallel.py:154]: Unclear whether the whole test or just the decorator should have been removed. The language of the test sounds nonspecific to the `auto_move_data` decorator

**Comment by Tshimanga** [comment]
From the DeepSource: Python check that failed
image

Did this check get added afterwards and only runs on files that have changes? Seems surprising this didn't come up earlier.

File: tests/core/test_decorators.py (hunk 1 above)
Comment (carmocca) [reply tests/core/test_decorators.py:18]: I think it's okay to delete the file, unless @kaushikb11 has in mind a possible test for it

File: tests/overrides/test_data_parallel.py (hunk 2 above)
Comment (carmocca) [reply tests/overrides/test_data_parallel.py:154]: I'd say what you did is correct :)

File: tests/core/test_decorators.py (hunk 1 above)
Comment (Tshimanga) [reply tests/core/test_decorators.py:18]: I'll remove for now and the file can be reinstated later

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by Borda — APPROVED** [review]
